pkg/kudoctl/cmd/test.go

```diff
@@ -0,0 +1,114 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+	"testing"
+
+	kudo "github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/kudobuilder/kudo/pkg/test"
+	testutils "github.com/kudobuilder/kudo/pkg/test/utils"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	testExample = `
+      Run tests configured by kudo-test.yaml:
+            kubectl kudo test
+
+      Load a specific test configuration:
+            kubectl kudo test --config test.yaml
+
+      Run tests against an existing Kubernetes cluster:
+            kubectl kudo test ./pkg/test/test_data/
+
+      Run tests against an existing Kubernetes cluster, and install KUDO, manifests, and CRDs for the tests:
+            kubectl kudo test --crd-dir ./config/crds/ --manifests-dir ./config/samples/test-framework/ ./pkg/test/test_data/
+
+      Run a Kubernetes control plane and KUDO and install manifests and CRDs for the running tests:
+            kubectl kudo test --start-control-plane --start-kudo --crd-dir ./config/crds/ --manifests-dir ./config/samples/test-framework/ ./pkg/test/test_data/
+`
+)
+
+// newTestCmd creates the test command for the CLI
+func newTestCmd() *cobra.Command {
+	configPath := ""
+	options := kudo.TestSuite{}
+	defaults := kudo.TestSuite{
+		TestDirs: []string{},
+	}
+
+	testCmd := &cobra.Command{
+		Use:   "test [flags]... [test directories]...",
+		Short: "-> Test KUDO and Frameworks.",
+		Long: `Runs integration tests against a Kubernetes cluster.
+
+The test framework supports connecting to an existing Kubernetes cluster or it can start a Kubernetes API server during the test run.
+
+It can also start up KUDO and apply manifests before running the tests.
+
+If no arguments are provided, the test harness will attempt to load the test configuration from kudo-test.yaml.
+
+For more detailed documentation, visit: https://kudo.dev/docs/testing`,
+		Example: testExample,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			options.TestDirs = args
+
+			if configPath == "" && reflect.DeepEqual(options, defaults) {
+				if _, err := os.Stat("kudo-test.yaml"); err == nil {
```

Comment:
I know this sounds silly and _shouldn't ever happen_, but I've done this incorrectly from the shell before (cp makes a dir instead of a file, or absentminded mkdir, or _whatever_), check that the fileinfo (what you have as `_` right now) is not a dir with fileInfo.IsDir(). :)

Reply:
I think it might actually be better to let it bubble up to the LoadYAML method:

```
➜  kudo git:(kudoctl-test) ✗ ./kudoctl test
Error: read kudo-test.yaml: is a directory
Usage:
  kubectl-kudo test [flags]... [test directories]...

Examples:

      Run tests configured by kudo-test.yaml:
            kubectl kudo test

      Load a specific test configuration:
            kubectl kudo test --config test.yaml

      Run tests against an existing Kubernetes cluster:
            kubectl kudo test ./pkg/test/test_data/

      Run tests against an existing Kubernetes cluster, and install KUDO, manifests, and CRDs for the tests:
            kubectl kudo test --crd-dir ./config/crds/ --manifests-dir ./config/samples/test-framework/ ./pkg/test/test_data/

      Run a Kubernetes control plane and KUDO and install manifests and CRDs for the running tests:
            kubectl kudo test --start-control-plane --start-kudo --crd-dir ./config/crds/ --manifests-dir ./config/samples/test-framework/ ./pkg/test/test_data/


Flags:
      --config string          Path to file to load test settings from (must not be set with any other arguments).
      --crd-dir string         Directory to load CustomResourceDefinitions from prior to running the tests.
  -h, --help                   help for test
      --manifests-dir string   A directory containing manifests to apply before running the tests.
      --start-control-plane    Start a local Kubernetes control plane for the tests (requires etcd and kube-apiserver binaries, implies --start-kudo).
      --start-kudo             Start KUDO during the test run.

➜  kudo git:(kudoctl-test) ✗ 
```

That way the error is more clear and we don't have to check every edge case here.

Reply:
:+1: sgtm